app/gui/view/graph-editor/src/selection/bounding_box.rs

```diff
@@ -74,6 +76,45 @@ impl BoundingBox {
         self.bottom -= size / 2.0;
         self.top += size / 2.0;
     }
+
+    /// Return the x position of the left boundary.
+    pub fn left(&self) -> f32 {
+        self.left
+    }
+
+    /// Return the x position of the right boundary.
+    pub fn right(&self) -> f32 {
+        self.right
+    }
+
+    /// Return the y position of the top boundary.
+    pub fn top(&self) -> f32 {
+        self.top
+    }
+
+    /// Return the y position of the bottom boundary.
+    pub fn bottom(&self) -> f32 {
+        self.bottom
+    }
+
+    /// Expand the boundaries to make them contain all points belonging to the bounding box passed
+    /// as an argument.
+    pub fn grow_to_include(&mut self, other: &BoundingBox) {
+        self.left = self.left.min(other.left);
+        self.right = self.right.max(other.right);
+        self.bottom = self.bottom.min(other.bottom);
+        self.top = self.top.max(other.top);
+    }
+
+    /// Calculates the squared norm of a vector between the point passed as argument, and a point
+    /// in the bounding box that is nearest to the point passed as an argument.
```

Comment:
The documentation does not mention if the distance is negative inside the bbox. In computer graphics its a popular solution for it to be negative inside.